parser/src/soft_keywords.rs

```diff
@@ -43,14 +46,16 @@ where
     fn next(&mut self) -> Option<LexResult> {
         let mut next = self.underlying.next();
         if let Some(Ok((tok, range))) = next.as_ref() {
-            // If the token is a `match` or `case` token, check if it's used as an identifier.
-            // We assume every `match` or `case` is an identifier unless both of the following
-            // conditions are met:
+            // If the token is a soft keyword e.g. `type`, `match`, or `case`, check if it's
+            // used as an identifier. We assume every soft keyword use is an identifier unless
+            // a heuristic is met.
+
+            // For `match` and `case`, all of the following conditions must be met:
             // 1. The token is at the start of a logical line.
             // 2. The logical line contains a top-level colon (that is, a colon that is not nested
             //    inside a parenthesized expression, list, or dictionary).
-            // 3. The top-level colon is not the immediate sibling of a `match` or `case` token.
-            //    (This is to avoid treating `match` and `case` as identifiers when annotated with
+            // 3. The top-level colon is not the immediate sibling of a soft keyword token.
+            //    (This is to avoid treating soft keywords as identifiers when annotated with
```

Comment:
```suggestion
            // 3. The top-level colon is not the immediate sibling of a `match` or `case` token.
            //    (This is to avoid treating `match` or `case` as identifiers when annotated with
```